File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* mergeTwoLists(struct ListNode* list1, struct ListNode* list2) {

    struct ListNode* list3=NULL,*tail;

    if(list1==NULL)
    {
        return list2;
    }
    if(list2==NULL)
    {
        return list1;
    }
    if(list1->val < list2->val)
    {
        list3=list1;
        tail=list1;
        list1=list1->next;
    }
    else
    {
        list3=list2;
        tail=list2;
        list2=list2->next;
    }

    while(list1!=NULL && list2 !=NULL)
    {
        if(list1->val<list2->val)
        {
            tail->next=list1;
            tail=tail->next;
            list1=list1->next;
        }
        else
        {
            tail->next=list2;
            tail=tail->next;
            list2=list2->next;
        }
    }
    if(list1==NULL)
    {
        tail->next=list2;
    }
    else
    {
        tail->next=list1;
    }

    return list3;



    
    
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeTwoLists(self, list1, list2):
        if list1 is None and list2 is None:
            return list1
        if list1 is None and list2 is not None:
            return list2
        if list1 is not None and list2 is None:
            return list1
        dummy=ListNode(0)
        tail=dummy
        temp1=list1
        temp2=list2
        while temp1 is not None and temp2 is not None:
            if temp1.val<=temp2.val:
                tail.next=temp1
                temp1=temp1.next
            else:
                tail.next=temp2
                temp2=temp2.next
            tail=tail.next
        if temp1 is not None:
            tail.next=temp1
        if temp2 is not None:
            tail.next=temp2
        return dummy.next

        